core/src/main/java/jenkins/views/Header.java

```diff
@@ -54,4 +63,28 @@ public static Header get() {
         return header.orElseGet(JenkinsHeader::new);
     }
 
+    /**
+     * @return a list of {@link Action} to show in the header, defaults to {@link hudson.model.RootAction} extensions
+     */
+    @Restricted(NoExternalUse.class)
+    public List<Action> getActions() {
+        // There's an issue where new actions (e.g. a new plugin installation) don't appear in the order
+        // of their ordinal annotation - to work around that we manually sort the list
+        Map<String, Double> rootActionsOrdinal = ExtensionList.lookup(RootAction.class)
+                .getComponents()
+                .stream()
+                .collect(Collectors.toMap(
+                        c -> c.getInstance().getClass().getName(),
```

Comment:
This would not work if `@Extension` is placed on a member other than a type (`class`), which would be unusual but is legal. Nor would it work for dynamically installed (`ExtensionList.add`) actions. Whatever you are trying to solve here, it sounds like the wrong way to do it (maybe there is some minor unfiled bug in `ExtensionList`).